config/routes.rb

```diff
@@ -1,29 +1,97 @@
 require 'sidekiq/web' if defined? Sidekiq
 require Rails.root.join('lib', 'router_constraints.rb')
 
-Ontohub::Application.routes.draw do
+Specroutes.define(Ontohub::Application.routes) do
 
   resources :filetypes, only: :create
 
   # IRI Routing #
   ###############
-  #
+  # as per Loc/Id definition
+
+  # Special (/ref-based) Loc/Id routes
+  specified_get '/ref/:reference/:repository_id/*locid' => 'ontologies#show',
+    as: :ontology_iri_versioned,
+    constraints: [
+      RefLocIdRouterConstraint.new(Ontology, ontology: :id),
+      MIMERouterConstraint.new('text/plain', 'text/html')
+    ]
+
+  # MMT-Support
+  specified_get '/ref/mmt/:repository_id/*path' => 'ontologies#show',
+    as: :ontology_iri_mmt,
+    constraints: [
+      MMTRouterConstraint.new(Ontology, ontology: :id),
+      MIMERouterConstraint.new('text/plain', 'text/html')
+    ]
+
+  specified_get '/ref/mmt/:repository_id/*path' => 'mappings#show',
+    as: :ontology_iri_mmt,
+    constraints: [
+      MMTRouterConstraint.new(Mapping, ontology: :ontology_id, element: :id),
+      MIMERouterConstraint.new('text/plain', 'text/html')
+    ]
+
+  specified_get '/ref/mmt/:repository_id/*path' => 'symbols#index',
+    as: :ontology_iri_mmt,
+    constraints: [
+      MMTRouterConstraint.new(OntologyMember::Symbol, ontology: :ontology_id),
+      MIMERouterConstraint.new('text/plain', 'text/html')
+    ]
+
+  specified_get '/ref/mmt/:repository_id/*path' => 'sentences#index',
+    as: :ontology_iri_mmt,
+    constraints: [
```

Comment:
Style/AlignHash: Align the elements of a hash literal if they span more than one line.